### ISSUE TITLE
Routing?

### ISSUE THREAD

**Issue by wordofchristian** [dabbott/react-express#22]
I'm wondering if a section on routing options would be in scope for this project. Seems like it would fit well before the "Data Management" section in the table of contents.

**Comment by dabbott** [comment]
Totally in scope! I think you're right about it fitting before the "Data Management" section. Great minds think alike ;) https://github.com/dabbott/react-express/blob/1e71b72493a98e28387f556b7bed59c96cc28219/src/utils/Sections.js#L203

@wordofchristian do you have any interest in writing one? I've only used React Router and am by no means an expert.

Seems like [React Router](http://cdn.rawgit.com/dabbott/react-native-web-player/gh-v1.10.0/index.html#platform=web&vendorComponents=%5B%5B%22react-router%22%2C%22https%3A%2F%2Fcdnjs.cloudflare.com%2Fajax%2Flibs%2Freact-router%2F4.1.1%2Freact-router.min.js%22%5D%2C%5B%22react-router-dom%22%2C%22https%3A%2F%2Fcdnjs.cloudflare.com%2Fajax%2Flibs%2Freact-router-dom%2F4.1.1%2Freact-router-dom.min.js%22%5D%5D&code=import%20React%2C%20%7B%20Component%20%7D%20from%20'react'%0Aimport%20%7B%20render%20%7D%20from%20'react-dom'%0Aimport%20%7B%0A%20%20BrowserRouter%20as%20Router%2C%0A%20%20Route%2C%0A%20%20Link%0A%7D%20from%20'react-router-dom'%0A%0Aconst%20BasicExample%20%3D%20()%20%3D%3E%20(%0A%20%20%3CRouter%3E%0A%20%20%20%20%3Cdiv%3E%0A%20%20%20%20%20%20%3Cul%3E%0A%20%20%20%20%20%20%20%20%3Cli%3E%3CLink%20to%3D%22%2F%22%3EHomhiliohe%3C%2FLink%3E%3C%2Fli%3E%0A%20%20%20%20%20%20%20%20%3Cli%3E%3CLink%20to%3D%22%2Fabout%22%3EAbout%3C%2FLink%3E%3C%2Fli%3E%0A%20%20%20%20%20%20%20%20%3Cli%3E%3CLink%20to%3D%22%2Ftopics%22%3ETopics%3C%2FLink%3E%3C%2Fli%3E%0A%20%20%20%20%20%20%3C%2Ful%3E%0A%0A%20%20%20%20%20%20%3Chr%2F%3E%0A%0A%20%20%20%20%20%20%3CRoute%20exact%20path%3D%22%2F%22%20component%3D%7BHome%7D%2F%3E%0A%20%20%20%20%20%20%3CRoute%20path%3D%22%2Fabout%22%20component%3D%7BAbout%7D%2F%3E%0A%20%20%20%20%20%20%3CRoute%20path%3D%22%2Ftopics%22%20component%3D%7BTopics%7D%2F%3E%0A%20%20%20%20%3C%2Fdiv%3E%0A%20%20%3C%2FRouter%3E%0A)%0A%0Aconst%20Home%20%3D%20()%20%3D%3E%20(%0A%20%20%3Cdiv%3E%0A%20%20%20%20%3Ch2%3EHome%3C%2Fh2%3E%0A%20%20%3C%2Fdiv%3E%0A)%0A%0Aconst%20About%20%3D%20()%20%3D%3E%20(%0A%20%20%3Cdiv%3E%0A%20%20%20%20%3Ch2%3EAbout%3C%2Fh2%3E%0A%20%20%3C%2Fdiv%3E%0A)%0A%0Aconst%20Topics%20%3D%20(%7B%20match%20%7D)%20%3D%3E%20(%0A%20%20%3Cdiv%3E%0A%20%20%20%20%3Ch2%3ETopics%3C%2Fh2%3E%0A%20%20%20%20%3Cul%3E%0A%20%20%20%20%20%20%3Cli%3E%0A%20%20%20%20%20%20%20%20%3CLink%20to%3D%7B%60%24%7Bmatch.url%7D%2Frendering%60%7D%3E%0A%20%20%20%20%20%20%20%20%20%20Rendering%20with%20React%0A%20%20%20%20%20%20%20%20%3C%2FLink%3E%0A%20%20%20%20%20%20%3C%2Fli%3E%0A%20%20%20%20%20%20%3Cli%3E%0A%20%20%20%20%20%20%20%20%3CLink%20to%3D%7B%60%24%7Bmatch.url%7D%2Fcomponents%60%7D%3E%0A%20%20%20%20%20%20%20%20%20%20Components%0A%20%20%20%20%20%20%20%20%3C%2FLink%3E%0A%20%20%20%20%20%20%3C%2Fli%3E%0A%20%20%20%20%20%20%3Cli%3E%0A%20%20%20%20%20%20%20%20%3CLink%20to%3D%7B%60%24%7Bmatch.url%7D%2Fprops-v-state%60%7D%3E%0A%20%20%20%20%20%20%20%20%20%20Props%20v.%20State%0A%20%20%20%20%20%20%20%20%3C%2FLink%3E%0A%20%20%20%20%20%20%3C%2Fli%3E%0A%20%20%20%20%3C%2Ful%3E%0A%0A%20%20%20%20%3CRoute%20path%3D%7B%60%24%7Bmatch.url%7D%2F%3AtopicId%60%7D%20component%3D%7BTopic%7D%2F%3E%0A%20%20%20%20%3CRoute%20exact%20path%3D%7Bmatch.url%7D%20render%3D%7B()%20%3D%3E%20(%0A%20%20%20%20%20%20%3Ch3%3EPlease%20select%20a%20topic.%3C%2Fh3%3E%0A%20%20%20%20)%7D%2F%3E%0A%20%20%3C%2Fdiv%3E%0A)%0A%0Aconst%20Topic%20%3D%20(%7B%20match%20%7D)%20%3D%3E%20(%0A%20%20%3Cdiv%3E%0A%20%20%20%20%3Ch3%3E%7Bmatch.params.topicId%7D%3C%2Fh3%3E%0A%20%20%3C%2Fdiv%3E%0A)%0A%0Arender(%3CBasicExample%20%2F%3E%2C%20document.querySelector('%23app'))%0A) works in the web player too, which is awesome.

**Comment by wordofchristian** [comment]
> @wordofchristian do you have any interest in writing one? I've only used React Router and am by no means an expert.

Yeah I can write this. I was curious to see it myself because I was wanting to see what the contemporary best practices were with regards to the interaction between redux and the router. That may be a question for another day. 

This section should probably just cover the basic example of using react router.
